source/game.broadcastmaterialsource.base.bmx

```diff
@@ -100,6 +100,32 @@ Type TBroadcastMaterialSourceBase Extends TNamedGameObject {_exposeToLua="select
 
 		Return effects.Copy()
 	End Method
+	
+	
+	Method GetEffectsList:TList(name:String)
+		if not effects then Return Null
+		Return effects.GetList(name)
+	End Method
+
+
+	Method GetEffectsCount:Int(name:String)
+		if Not effects Then Return 0
+		Local l:TList = effects.GetList(name)
+		If l Then Return l.Count()
+		Return 0
+	End Method
+
+
+	Method GetEffects:TGameModifierGroup(createIfMissing:Int = True)
+		if not effects and createIfMissing Then effects = New TGameModifierGroup
+		Return effects
+	End Method
+	
+	
+	Method UpdateEffects:Int(name:String, params:TData)
+		If Not effects Then Return False
```

Comment:
Sollte in dem Fall nicht immer ein Effekt-Objekt angelegt werden?

Reply:
Nein.
Es soll ja nur ein Update ausfuehren, wenn irgendwas einen Modifier dafuer hinterlegt hat.

Reply:
Eventuell verwechselst Du es mit den `Update()` der anderen Modifierklassen.

Die von `TGameModifierGroup` ist:
```
	Method Update:Int(trigger:String, params:TData)
		Local l:TList = GetList(trigger)
		If Not l Then Return 0

		For Local modifier:TGameModifierBase = EachIn l
			modifier.Update(params)
		Next

		Return l.Count()
	End Method
```
und da wird nur Zeugs innerhalb einer Liste aktualisiert - und die Liste ist dem `trigger` (also bspweise "happen") zugeordnet.